### PR TITLE
Allow reading from ECS container credentials

### PR DESCRIPTION
**Pull request by jon-mago**
This is my attempt at fixing #23 .

I'm not sure this is the best place to put the code, or if it should exist in `aws.ec2metadata` package. Other things:

1. I've not yet updated `DESCRIPTION` or `NEWS.md`
2. I wasn't sure how to test this

Please ensure the following before submitting a PR:

 - [x] if suggesting code changes or improvements, [open an issue](https://github.com/cloudyr/aws.signature/issues/new) first
 - [ ] for all but trivial changes (e.g., typo fixes), add your name to [DESCRIPTION](https://github.com/cloudyr/aws.signature/blob/master/DESCRIPTION)
 - [x] for all but trivial changes (e.g., typo fixes), documentation your change in [NEWS.md](https://github.com/cloudyr/aws.signature/blob/master/NEWS.md) with a parenthetical reference to the issue number being addressed
 - [x] if changing documentation, edit files in `/R` not `/man` and run `devtools::document()` to update documentation
 - [ ] add code or new test files to [`/tests`](https://github.com/cloudyr/aws.signature/tree/master/tests/testthat) for any new functionality or bug fix
 - [x] make sure `R CMD check` runs without error before submitting the PR

### CODE REVIEW

**Comment by leeper** [comment]
Thanks. I think this would be better as two separate PRs:

1. A PR against [aws.ec2metadata](https://github.com/cloudyr/aws.ec2metadata) that actually parses the credentials. This could be a function `is_ecs()` (like the current `is_ec2()`) that provides a check for whether we're in an ECS container and then a separate function (`container_metadata()` to return credentials.
2. A PR here that adds a layer analogous to the current `is_ec2()` check, and if we're there, then pulls credentials from the `container_metadata()` function.

That keeps functionality consistent and avoids having to import jsonlite and curl here.

**Comment by leeper** [comment]
I've merged in https://github.com/cloudyr/aws.ec2metadata/pull/2, so please feel free to update this PR accordingly. Thanks, again!

**Comment by jon-mago** [comment]
Hi

I updated this PR too.  The travis check is failing because the new ec2metadata package is not yet available for it...

**Comment by jon-mago** [comment]
Hi, just poking about this again.  As I said, I can't get travis to pass until the new ec2metadata is available (I think, I don't have much travis experience)

File: R/locate_credentials.R [hunk 1]
```diff
@@ -184,32 +219,7 @@ function(key = NULL,
...
-                    message(sprintf("Using EC2 Instance Metadata for AWS Region ('%s')", region))
-                }
-            } else {
-                region <- default_region
-                if (isTRUE(verbose)) {
-                    message(sprintf("Using default value for AWS Region ('%s')", region))
-                }
-            }
-        }
+        region <- get_region(region, default_region)
```
Comment (leeper) [inline R/locate_credentials.R]: Can you undo this change? I don't think we need this as a separate function (and the function as proposed won't work because it doesn't pass the `env` variable forward).

**Comment by jon-mago** [comment]
I'd missed that that would break, sorry.  I wanted to avoid having two nearly identical blocks for looking up the region for ECS and EC2.  I'll rethink and revise, though

**Comment by leeper** [comment]
No worries!
